modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/utils/JavaParserUtils.java

```diff
@@ -206,29 +208,24 @@ public static MuleVersion calculateConnectionProviderMinMuleVersion(Type connect
    * @param source the connection provider, as a {@link Type}, to calculate its min mule version
    * @return the minimum mule version of the given source
    */
-  public static MuleVersion calculateSourceMinMuleVersion(Type source) {
+  public static MuleVersion calculateSourceMinMuleVersion(SourceElement source) {
+    MuleVersion sourceGenericsMMV = calculateSourceGenericsMinMuleVersion(source.getSuperClassGenerics());
```

Comment:
Have you checked that this issue does not happen with other cases, like connection providers?

Reply:
Yes. right now for connection providers we are not analyzing its generic.